README.md

```diff
@@ -1,2 +1,103 @@
+
+### Running the sample
+
+#### Preparation
+1) Create the target container
+Create the container the data should be ingested in. Make sure the number of pyhsical partitions created is high enough to accomodate the data being ingested. The explanation how ti achieve this is located [here](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos-spark_3_2-12/docs/scenarios/Ingestion.md#creating-a-new-container-if-the-ingestion-via-the-cosmos-spark-connector-is-for-the-initial-migration) - 
+The sample expects that the container uses `/id` as partition key - if you want to use a different partition key, some code changes will be required first. When creating the container please ensure the partition key definition is set to `/id` and the indexing policy is updated to explude as many properties aspossible (each to-be-indexed property will add to the RU charge for write operations, so the fewer properties are indexed, the lower the RU charge for each write operation).
+The ideal index policy for the target container would look like this:
+
+```xml
+{
+    "indexingMode": "consistent",
+    "automatic": true,
+    "includedPaths": [
+        
+    ],
+    "excludedPaths": [
+        {
+            "path": "/*"
+        },
+        {
+            "path": "/\"_etag\"/?"
+        }
+    ]
+}
+```
+
+2) Create the Job contianer to use for tracking status etc. with metadata documents. This container has to use a partition key definition of `/pk` - and the standard indexing policy should be used.
+
+3) Sepcify the environment variables according to the (Configuration)[#Configuration] section above.
+
+#### Building the application
+You can use `mvn install` to build the jar file. The application will produce a fat-jar (one single jar file containing all dependencies). This file will be located in the traget folder and have a name similar to `azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar`.
+
+#### Creating a new ingestion job
+Each ingestion will be tracked in a `Job` - the job and it's batches (fragments of input files) is tracked in a Cosmos DB Container used to store the metadata. The workers will then load-balance the ingestion across these batches.
+
+To create a new ingestion job - run the following command line
+```
+java -cp ./azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar com.azure.cosmos.samples.distributedbulk.Main create SampleJob01 ^.*\.json$
+```
+
+A slightly different syntax can be used to create the job quicker when you know that all your input files have the same number of lines/documents. In this case not each file is downloaded form Storage to count the target number of lines - but this step is done only for one file and then the same numbe rof lines is used for all files.
+```
+java -cp ./azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar com.azure.cosmos.samples.distributedbulk.Main createUnifrom SampleJob01 ^.*\.json$
```

Comment:
I didn't fully grasp the use case of `createUniform`.

Reply:
Added more verbose explanation - hopefully addresses this.